sphinx_gallery/sorting.py

```diff
@@ -46,3 +48,71 @@ def __call__(self, item):
             raise ValueError('If you use an explicit folder ordering, you '
                              'must specify all folders. Explicit order not '
                              'found for {}'.format(item))
+
+
+class _SortKey(object):
+    """Base class for section order key classes."""
+
+    def __init__(self, src_dir):
+        self.src_dir = src_dir
+
+
+class AmountOfCodeSortKey(_SortKey):
+    """Sort examples in src_dir by amount of code.
+
+    Parameters
+    ----------
+    src_dir : str
+        The source directory.
+    """
+
+    def __call__(self, filename):
+        src_file = os.path.normpath(os.path.join(self.src_dir, filename))
+        file_conf, script_blocks = split_code_and_text_blocks(src_file)
```

Comment:
ah I see that it isn't strictly `NumberOfLines` per-se, since you're only counting the code lines. I think it's fine to keep it `AmountOfCode` in that case, but maybe clarify in the docs that this will count only code lines and not all the lines in the `.py` files